spanner/read.go

```diff
@@ -672,7 +676,19 @@ func (d *resumableStreamDecoder) tryRecv(mt *builtinMetricsTracer, retryer gax.R
 	if d.err == nil {
 		d.q.push(res)
 		if res.GetLast() {
-			d.changeState(finished)
+			d.changeState(ended)
+			go func(s streamingReceiver) {
+				for {
+					_, err := s.Recv()
+					if err == io.EOF {
+						break
+					}
+					if err != nil {
+						break
+					}
+				}
+
```

Comment:
I would suggest that we simplify this to only `_, _ = s.Recv()`. The reason is that:
1. The contract is that we only reach this if `Last` has been set. That means that calling `s.Recv()` once should be enough to get the trailers and end the stream.
2. If the server for whatever reason does not comply with the contract and returns `Last=true` before the end of the stream, we are still stopping early. However, this would then at least cause the stream to marked as cancelled instead of ended normally.